docs/site/todo-list-tutorial-controller.md

```diff
@@ -0,0 +1,169 @@
+---
+lang: en
+title: "Add TodoList and TodoList's Todo Controller"
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/todo-list-tutorial-controller.html
+summary: LoopBack 4 TodoList Application Tutorial - Add TodoList and TodoList's Todo Controller
+---
+
+### Controllers with related models
+
+Defining business logic to handle requests to related models isn't too different
+from handling requests to standalone models. We'll create controllers to handle
+requests for todo-lists and todo items under a todo-list.
+
+### Create TodoList controller
+
+Starting with creating a controller for our `TodoList` routes, run the CLI
+command for creating a RESTful CRUD controller with the following inputs:
+
+```sh
+$ lb4 controller
+? Controller class name: TodoList
+? What kind of controller would you like to generate? REST Controller with CRUD functions
+? What is the name of the model to use with this CRUD repository? TodoList
+? What is the name of your CRUD repository? TodoRepository
+? What is the type of your ID? number
+? What is the base HTTP path name of the CRUD operations? /todo-lists
+   create src/controllers/todo-list.controller.ts
+   update src/controllers/index.ts
+
+Controller TodoList was created in src/controllers/
+```
+
+And voilà! We now have a set of basic APIs for todo-lists just like that!
+
+### Create TodoList's Todo controller
+
+For the controller handling `Todos` of a `TodoList`, we'll start with an empty
+controller:
+
+```sh
+$ lb4 controller
+? Controller class name: TodoListTodo
```

Comment:
Is this the best practice we want to promote? ... a new controller for the relation APIs? If I was writing this application since the base path for this is `/todo-lists`, I would add the new controller methods to the same file as `TodoListController` instead of `TodoListTodoController`.

Reply:
> Is this the best practice we want to promote? ... a new controller for the relation APIs?

Yes, each relation should have its own controller.

While it may be tempting to add relation API methods to the same controller which provide regular CRUD methods, this approach does not scale well as more relations are added to the original model.

Having a new controller for each relation makes it also easier for our CLI to scaffold REST API implementation - we don't need to modify any existing controller code, just generate a new controller file.